flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/CliClient.java

```diff
@@ -624,4 +637,56 @@ private static Terminal createDefaultTerminal() {
 			throw new SqlClientException("Error opening command line interface.", e);
 		}
 	}
+
+	/**
+	 * Split a SemiColon-separated String, but ignore SemiColons in quotes.
+	 */
+	static List<String> splitSemiColon(String line) {
+		boolean inSingleQuotes = false;
+		boolean inDoubleQuotes = false;
+		boolean escape = false;
+
+		// normalize
+		line = line.replaceAll("--[^\r\n]*", ""); // remove single-line comments
+		line = line.replaceAll("/\\*[\\w\\W]*?(?=\\*/)\\*/", ""); // remove double-line comments
```

Comment:
This is the another concern from my side. We will lose the original text and character positions after replacing. 
This will make users confused if the SQL has parser error because the position is not correct. 

Another option is support all parsers in `flink-sql-parser`. The sql parser supports multi statement naturally and have proper exceptions. In this way, we will not need this "hack" replacing. But we may need more works. 

cc @danny0405 

Besides of that, we have met some issues in internal blink when removing comments and

Reply:
Support multiline sql parse work for sql-parser seems the right way to go, just one question: after parsing, we got a `SqlNodeList` with each node comes from a sql statement.

So we may need to unparse the node to sql text again when passing them to executor. The unparsed text may be different from the original one, so the error message may not be exact accurate sometimes.

Reply:
If it is an invalid multiline sql string, then it should fail in sql parser instead of passing to executor?

Reply:
Yeah, we should  just to keep the unparsed sql correct.